### ISSUE TITLE
Run detekt and ktlint on sample projects

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#257]
I just noticed that we do not run ktlint and detekt on our samples. In [build.gradle](https://github.com/mozilla-mobile/android-components/blob/master/build.gradle) both tools are configured to only look into `components/*`

**Comment by pocmo** [comment]
detekt:

```
Ruleset: comments
Ruleset: complexity
        LongMethod - 29/20 - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:183:5
        ComplexMethod - 10/10 - [simulateReload] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:364:5
Ruleset: empty-blocks
Ruleset: exceptions
        TooGenericExceptionCaught - [t] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:384:22
Ruleset: naming
Ruleset: performance
Ruleset: potential-bugs
Ruleset: style
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:191:36
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:192:50
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:194:28
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:195:28
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:196:28
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:197:28
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:206:35
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:207:36
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:210:50
        MagicNumber - [setupSeedlingToolbar] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:220:43
        MagicNumber - [space] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:306:52
        MagicNumber - [loop] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:371:43
        MagicNumber - [loop] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:371:52
        MagicNumber - [loop] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:382:38
        MagicNumber - [onBindViewHolder] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:51:49
        MagicNumber - [progress] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:92:40
        MagicNumber - [progress] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:92:47
        MagicNumber - [progress] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:93:38
        MagicNumber - [progress] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:100:26
        MagicNumber - [<anonymous>] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:102:27
        MagicNumber - [<anonymous>] at android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/SampleToolbarHelpers.kt:112:17
```

ktlint:

```
/Users/skaspari/Projects/Mozilla/android-components/samples/browser/src/main/java/org/mozilla/samples/browser/Components.kt:32:16: Unexpected spacing before ":"

**Comment by ali-abdullah** [comment]
Hey, @pocmo I was wondering if I could work on this issue and #976 since they're similar. 
I've already fixed the ktlint issues on my local fork, working on the detekt issues. 

Also, our `detekt` log looks similar but I got a lot more issues for `ktlint`

**Comment by ali-abdullah** [comment]
my output for ktlint, I fixed these linting issues though
```
/Users/aliabdullah/side/android-components/samples/browser/src/geckoBeta/java/org/mozilla/samples/browser/Components.kt:11:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/browser/src/geckoBeta/java/org/mozilla/samples/browser/Components.kt:17:86: Missing spacing before "{" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/browser/src/geckoNightly/java/org/mozilla/samples/browser/Components.kt:10:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/browser/src/geckoRelease/java/org/mozilla/samples/browser/Components.kt:10:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/browser/src/geckoRelease/java/org/mozilla/samples/browser/Components.kt:16:98: Missing spacing before "{" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/browser/src/main/java/org/mozilla/samples/browser/IntentReceiverActivity.kt:31:1: Unexpected blank line(s) before "}" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:14:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:23:51: Missing spacing after "{" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:23:71: Missing spacing around "+" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:59:32: Redundant curly braces (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:63:32: Redundant curly braces (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/MainActivity.kt:63:60: Redundant curly braces (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/ProtectedDataAdapter.kt:13:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/ProtectedDataAdapter.kt:19:9: Unexpected indentation (expected 4, actual 8) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/ProtectedDataAdapter.kt:20:9: Unexpected indentation (expected 4, actual 8) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/ProtectedDataAdapter.kt:21:9: Unexpected indentation (expected 4, actual 8) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/dataprotect/src/main/java/org/mozilla/samples/dataprotect/ProtectedDataAdapter.kt:52:33: Missing spacing before ":" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/LoginFragment.kt:36:31: Parameter should be on a separate line (unless all parameters can fit a single line) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/LoginFragment.kt:36:57: Parameter should be on a separate line (unless all parameters can fit a single line) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/LoginFragment.kt:37:31: Unexpected indentation (expected 8, actual 30) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/LoginFragment.kt:37:58: Missing newline before ")" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt:18:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt:64:1: Unexpected blank line(s) before "}" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/ScanActivity.kt:13:74: Unnecessary space(s) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/LoginFragment.kt:36:31: Parameter should be on a separate line (unless all parameters can fit a single line) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/LoginFragment.kt:36:57: Parameter should be on a separate line (unless all parameters can fit a single line) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/LoginFragment.kt:37:31: Unexpected indentation (expected 8, actual 30) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/LoginFragment.kt:37:58: Missing newline before ")" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:17:1: Unused import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:18:1: Wildcard import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:50:31: Unnecessary semicolon (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:69:1: Unexpected blank line(s) before "}" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:70:1: Needless blank line(s) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:123:75: Unnecessary semicolon (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/MainActivity.kt:124:55: Unnecessary semicolon (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/SyncLoginsClient.kt:3:1: Wildcard import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/SyncLoginsClient.kt:5:1: Wildcard import (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/SyncLoginsClient.kt:88:21: Unnecessary space(s) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/SyncLoginsClient.kt:90:25: Unnecessary space(s) (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/sync-logins/src/main/java/org/mozilla/samples/sync/logins/SyncLoginsClient.kt:101:1: Unexpected blank line(s) before "}" (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:83:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:85:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:90:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:92:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:110:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:112:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:121:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:123:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:133:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:135:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:144:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:146:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:151:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:153:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:169:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:171:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:185:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:187:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:197:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:199:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:208:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:210:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:220:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:222:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:233:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:236:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:243:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:245:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:280:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:282:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:323:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:325:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:335:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:337:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:344:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:346:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:351:9: Missing space after // (cannot be auto-corrected)
/Users/aliabdullah/side/android-components/samples/toolbar/src/main/java/org/mozilla/samples/toolbar/ToolbarActivity.kt:353:9: Missing space after // (cannot be auto-corrected)
```

**Comment by pocmo** [comment]
@ali-abdullah Yeah, you can work on that and this would be super helpful! :)

Feel free to open a PR for only a subset of fixes (easier to review and land!) and then later other PRs for the rest. If a single PR gets too big then it's hard to review and may cause conflicts with other changes. :)

> Also, our detekt log look similar but I got a lot more issues for ktlint

That's absolutely possible. We added a bunch of more code to the sample apps since then.

**Comment by ali-abdullah** [comment]
@pocmo I can't seem to setup multiple inputs for `detekt` 
I've tried `files("$projectDir/components", "$projectDir/samples")` , digged a little deeper and apparently it was fixed for version `RC9.2` https://github.com/arturbosch/detekt/pull/1134 
but I'm not sure when it was broken. 

I then tried using profiles and creating a separate profile for `samples` but couldn't execute it from the terminal, I believe the command is `gradle detektCheck -Ddetekt.profile=[samples]` based on `--help` and release notes. 
``` 
    profile("samples") {
        input = "$projectDir/samples"
        config = "$projectDir/config/detekt.yml"
        filters = ".*test.*,.*/resources/.*,.*/tmp/.*"
        output = "$projectDir/build/reports/detekt"
        baseline = "$projectDir/config/detekt-baseline.xml"
    }
```
tl;dr : how do you get detekt to input multiple files or how do we execute different profiles

**Comment by ali-abdullah** [comment]
also, we could bump up the detekt version but that would require a gradle version bump too

**Comment by pocmo** [comment]
> also, we could bump up the detekt version but that would require a gradle version bump too

This sounds like something we probably want to do. I already have PR that starts to update the gradle version. Can you file a separate issue for updating detekt? :)

I think back then I just configured detekt to run on $projectDir.

**Comment by ali-abdullah** [comment]
Created issue here https://github.com/mozilla-mobile/android-components/issues/1052
